doc/source/coregistration.md

```diff
@@ -16,250 +16,472 @@ kernelspec:
+from matplotlib import pyplot
+pyplot.rcParams['figure.dpi'] = 600
+pyplot.rcParams['savefig.dpi'] = 600
+pyplot.rcParams['font.size'] = 9  # Default 10 is a bit too big for coregistration plots
+```
 
-**Example data**
+```{code-cell} ipython3
+import xdem
 
-Examples are given using data close to Longyearbyen on Svalbard. These can be loaded as:
+# Create a coregistration pipeline
+my_coreg_pipeline = xdem.coreg.ICP() + xdem.coreg.NuthKaab()
+
+# Or use a single method
+my_coreg_pipeline = xdem.coreg.NuthKaab()
+```
+
+Then, coregistering a pair of elevation data can be done by calling {func}`xdem.DEM.coregister_3d` from the DEM that should be aligned.
 
 ```{code-cell} ipython3
+:tags: [hide-cell]
+:mystnb:
+:  code_prompt_show: "Show the code for opening example files"
+:  code_prompt_hide: "Hide the code for opening example files"
+
 import geoutils as gu
 import numpy as np
+import matplotlib.pyplot as plt
 
-import xdem
-
-# Open a reference DEM from 2009
+# Open a reference and to-be-aligned DEM
 ref_dem = xdem.DEM(xdem.examples.get_path("longyearbyen_ref_dem"))
-# Open a to-be-aligned DEM from 1990
-tba_dem = xdem.DEM(xdem.examples.get_path("longyearbyen_tba_dem")).reproject(ref_dem, silent=True)
+tba_dem = xdem.DEM(xdem.examples.get_path("longyearbyen_tba_dem"))
+```
 
-# Open glacier polygons from 1990, corresponding to unstable ground
-glacier_outlines = gu.Vector(xdem.examples.get_path("longyearbyen_glacier_outlines"))
-# Create an inlier mask of terrain outside the glacier polygons
-inlier_mask = glacier_outlines.create_mask(ref_dem)
+```{code-cell} ipython3
+# Coregister by calling the DEM method
+aligned_dem = tba_dem.coregister_3d(ref_dem, my_coreg_pipeline)
 ```
 
-(coreg_object)=
-## The {class}`~xdem.Coreg` object
+Alternatively, the coregistration can be applied by calling {func}`~xdem.coreg.Coreg.fit_and_apply`, or sequentially
+calling the {func}`~xdem.coreg.Coreg.fit` and {func}`~xdem.coreg.Coreg.apply` steps,
+which allows a broader variety of arguments at each step, and re-using the same transformation to several objects
+(e.g., horizontal shift of both a stereo DEM and its ortho-image).
+
+```{code-cell} ipython3
+# (Equivalent) Or use fit and apply
+aligned_dem = my_coreg_pipeline.fit_and_apply(ref_dem, tba_dem)
+```
 
-Each coregistration approach in xDEM inherits their interface from the {class}`~xdem.Coreg` class<sup>1</sup>.
+Information about the coregistration inputs and outputs is summarized in {func}`~xdem.coreg.Coreg.info`.
 
-```{margin}
-<sup>1</sup>In a style resembling [scikit-learn's pipelines](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html#sklearn-linear-model-linearregression).
+```{tip}
+Often, an `inlier_mask` has to be passed to {func}`~xdem.coreg.Coreg.fit` to isolate static surfaces to utilize during coregistration (for instance removing vegetation, snow, glaciers). This mask can be easily derived using {func}`~geoutils.Vector.create_mask`.
```

Comment:
I would recommend using the function `create_inlier_mask` from the workflow submodule.

Reply:
This doesn't seem to have been limiting to users (and `workflows` is not in the API yet). Maybe we should discuss on the organization of this separately once you're back, also in relation to the CLI for coregistration.